[댓글 기능] 올레 미션 제출합니다.

jinwookh opened the pull request.

jinwookh committed:
refactor: use httpSession directly instead using request

jinwookh committed:
refactor: use optional at findByEmail

jinwookh committed:
refactor: move function of matching password to User from LoginService

jinwookh committed:
refactor: use RedirectionView object for redirection

jinwookh committed:
refactor: use User parameter at delete and save method

jinwookh committed:
refactor: delete loginDTO and use UserDTO instead

jinwookh committed:
refactor: delete update method that use jpql query in UserRepository

use update function of User and add Transactiional annotation instead

jinwookh committed:
refactor: delete update method that use jpql query in ArticleRepository

use update function of Article and add Transactiional annotation instead

jinwookh committed:
feat: add reference of article and user entity to each other

jinwookh committed:
feat: add new ArrayList at articles field in User class

jinwookh committed:
feat: add column length constraint at User

jinwookh committed:
docs: write todo file by content of 3rd mission

jinwookh committed:
docs: add 3rd mission required to readme file

jinwookh committed:
docs: add check markdown to todo file

jinwookh committed:
fix: bug that password checking logic does not work

change password to confirmpasssword at signup.html

jinwookh committed:
feat: comment and comment repository

add test case of comment repository's CRUD method

jinwookh committed:
feat: create comment and show comments

jinwookh committed:
feat: delete comment and show comment count at index page

jinwookh committed:
refactor: change request mapping parameter

remove duplicated url

jinwookh committed:
bug: comment editting window is not shown at article page

jinwookh committed:
fix: previous bug fixed

405 not found bug remained

jinwookh committed:
fix: 405 not found of put method(comment edit) bug fixed

jinwookh committed:
feat: alert error message at article page

jinwookh committed:
feat: block unauthorized author of article

jinwookh committed:
refactor: use equals method at comparing user instead using '=='

jinwookh committed:
refactor: add cascade option at @ManyToOne at Article

jinwookh committed:
refactor: put user object into article-edit html at createform method in ArticleController class

jinwookh committed:
refactor: change method name of controller

jinwookh committed:
refactor: delete ArticleController test file and CommentController test file

jinwookh committed:
refactor: add ArticleControllerTest

still some of tests fails

kang-hyungu reviewed (changes requested):
안녕하세요 올레님 리뷰어 강현구입니다.
댓글 기능 추가하신 내용 확인했어요
간단하게 몇 가지 코멘트 남겼으니 확인해보시고 반영해보아요!

jinwookh committed:
refactor: tests of ArticleControllerTest all success

jinwookh committed:
refactor: use nested tests at ArticleControllerTest

still editting feature test fails

jinwookh committed:
refactor: check whether user is author when user tries to edit

test all success

jinwookh committed:
refactor: delete setAuthor in Article class and add user parameter at ArticleDTO.toDomain method

jinwookh committed:
refactor: add isAuthorOf method at user class

checks whether user is author of article parameter

jinwookh committed:
refactor: delete name variable of PathVariable annotation

jinwookh committed:
refactor: add user in model, at get request mapping of article edit page

jinwookh committed:
refactor: delete setter method at comment and add parameters to CommentDTO.toDomain method

jinwookh committed:
refactor: move finding article method code right above of article using code

jinwookh committed:
refactor: add error logger at exception handling method

jinwookh committed:
refactor: add CommentController tests at CommentControllerTest file

jinwookh committed:
refactor: set message string field at NotFoundCommentException and UnAuthenticatedUserException

jinwookh committed:
refactor: add code that checks whether user exists at LoginService.getLoginUser method

set message string field at NotFoundUserException

jinwookh committed:
refactor: delete articleRepository tests and commentRepository tests

jinwookh committed:
refactor: add lazy fetching at ManyToOne annotation at entity classes

delete private access of noArgsConstructor, because it gives you error when you do lazy fetching

kang-hyungu reviewed (approved):
피드백 반영 잘 하셨어요 👍 
jpa 지연로딩 사용해보고 어떤 장점이 있는지도 잘 파악하셨어요
작업하신 내용은 머지할게요! 수고하셨어요!!
